Azure/sonic-buildimage
# [build]: Add possibility to override dockerd config file for multiarch builds.

#### Why I did it
Fixes #8898 

Dockerd for multiarch build by default use host OS config file ("/etc/docker/daemon.json").
Default configuration used on host OS may not work for containers run inside sonic-slave.
"DOCKER_CONFIG_FILE_FOR_MULTIARCH" variable allows overriding path to the
config file that will be used for multiarch dockerd.

#### How I did it

Added "DOCKER_CONFIG_FILE_FOR_MULTIARCH" to Makefile.work file that allow to
override path to dockerd config file through environment variable:
```
DOCKER_CONFIG_FILE_FOR_MULTIARCH=${path_to_file}/daemon.json   make ...
```
If the env variable is not set build the system preserves its default behavior.

#### How to verify it
- Set DOCKER_CONFIG_FILE_FOR_MULTIARCH env variable
- Run build
- While build is running execute `ps -eo pid,cmd | grep "[0-9] dockerd.*march"` command
- Verify that `--config-file` parameter is set to the same path that was specified in DOCKER_CONFIG_FILE_FOR_MULTIARCH variable.

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog



#### A picture of a cute animal (not mandatory but encouraged)
